File: BE/ClinicBooking/Models/Doctor.cs
namespace ClinicBooking.Models
{
    public class Doctor : EntityBase
    {
        public int DepartmentID { get; set; }
        public Department Department { get; set; }
        public string Certificate { get; set; }

        public int UserId { get; set; }

        public User User { get; set; }
    }
}

File: fe/types/doctor.ts
export interface DepartmentDTO {
  id: number;
  name: string;
}

export interface DoctorDTO {
  id: number;
  departmentID: number;
  department: DepartmentDTO;
  certificate: string;
  userId: number;
  firstName: string;
  lastName: string;
  active: boolean;
}

export interface Pagination {
  pageSize: number;
  pageNumber: number;
  totalItems: number;
  totalPages: number;
}

export interface DoctorsResponse {
  status: number;
  message: string;
  data: DoctorDTO[];
  pagination: Pagination;
}
